sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -31,15 +31,27 @@ class RapidsHostMemoryStore(
     extends RapidsBufferStore("host", catalog) {
   private[this] val pool = HostMemoryBuffer.allocate(maxSize, false)
   private[this] val addressAllocator = new AddressSpaceAllocator(maxSize)
+  private[this] var haveLoggedMaxExceeded = false
 
-  // Returns an allocated host buffer and whether the allocation is from the pinned pool
+  // Returns an allocated host buffer and whether the allocation is from the internal pool
   private def allocateHostBuffer(size: Long): (HostMemoryBuffer, Boolean) = {
-    require(size <= maxSize, s"allocating a buffer of $size bytes beyond max of $maxSize")
+    // spill to keep within the targeted size
+    synchronousSpill(math.max(maxSize - size, 0))
+
     var buffer: HostMemoryBuffer = null
     while (buffer == null) {
       buffer = PinnedMemoryPool.tryAllocate(size)
       if (buffer != null) {
-        return (buffer, true)
+        return (buffer, false)
```

Comment:
seems like this should return `true` here? It got a buffer from the pool if I read it correctly.

Reply:
Not the internal pool maintained by the host spill store but the external pinned memory pool.  Essentially this flag is for tracking if it needs to release it with the address allocator when the buffer is freed.

Reply:
ok got it.. the rest get freed via ref count

Reply:
The perhaps we need some better comments in allocateHostBuffer to explain this.